sip-core/src/main/java/de/ikor/sip/foundation/core/trace/CustomTracer.java

```diff
@@ -18,35 +18,40 @@ public class CustomTracer extends DefaultTracer {
 
   public static final String TRACING_ID = "tracingId";
 
-  private final Set<SIPTraceOperation> sipTraceOperations;
-
-  private final TraceHistory traceHistory;
+  private final SIPTraceConfig sipTraceConfig;
 
   /**
    * Creates new instance of CustomTracer Enables tracing in CamelContext
    *
-   * @param traceHistory {@link TraceHistory}
    * @param exchangeFormatter {@link SIPExchangeFormatter}
    * @param camelContext {@link CamelContext}
-   * @param sipTraceOperations set of {@link SIPTraceOperation}
+   * @param sipTraceConfig set of {@link SIPTraceConfig}
    */
   public CustomTracer(
-      TraceHistory traceHistory,
       SIPExchangeFormatter exchangeFormatter,
       CamelContext camelContext,
-      Set<SIPTraceOperation> sipTraceOperations) {
+      SIPTraceConfig sipTraceConfig) {
     setExchangeFormatter(exchangeFormatter);
     camelContext.setTracing(true);
-    this.traceHistory = traceHistory;
-    this.sipTraceOperations = sipTraceOperations;
+    this.sipTraceConfig = sipTraceConfig;
+  }
+
+  @Override
+  public void traceBeforeRoute(NamedRoute route, Exchange exchange) {
+    setTracingID(exchange);
+    super.traceBeforeRoute(route, exchange);
   }
 
   @Override
   public void traceBeforeNode(NamedNode node, Exchange exchange) {
-    exchange.getIn().setHeader(TRACING_ID, tracingList(exchange));
+    setTracingID(exchange);
     super.traceBeforeNode(node, exchange);
   }
 
+  private void setTracingID(Exchange exchange) {
+    exchange.getIn().setHeader(TRACING_ID, tracingList(exchange));
+  }
+
   private String tracingList(Exchange exchange) {
     String list = exchange.getIn().getHeader(TRACING_ID, String.class);
     if (list == null) {
```

Comment:
I know it's not in scope of this PR, but I think we should use some concrete object instead of list. 
This object should encapsulate the list and expose only operations we want to support.
It should know how to serialize itself into csv format.
Perhaps it should also be able to instantiate itself statically with Exchange parameter.

Reply:
I agree